packages/flutter_driver/lib/src/extension/web_extension.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:html' as html;
+import 'dart:js';
+import 'dart:js_util' as js_util;
+
+/// The dart:html implementation of [registerWebServiceExtension].
+///
+/// Registers Web Service Extension for Flutter Web application.
+///
+/// window.$flutterDriver will be called by Flutter Web Driver to process
+/// Flutter Command.
+///
+/// See also:
+///
+///  * [io_extension.dart], which has the dart:io implemeantion
+void registerWebServiceExtension(Future<Map<String, dynamic>> Function(Map<String, String>) call) {
```

Comment:
This needs a unit test for web non-web. In non-web, expect the thrown UnsupportedError, in web expect the property on the window.